### DISCUSSION TITLE
Support group by time interval with transformations

### DISCUSSION THREAD

**Discussion by woodsaj** [grafana/grafana#38169] [Dashboards, Panels & Visualizations]
I would like to be able to group input data by a time interval.
eg, given input data that looks like

timestamp | name | value
--------------|----------|-------- 
2020-10-29 16:28:31 | foo | 10
2020-10-29 16:28:12 | foo | 8
2020-10-29 16:25:37 | foo | 1

I want to be able to group the data into time buckets and select the aggregation to apply to each bucket.

eg, group into 1min buckets, and for each 'name' get the sum of 'value'

timestamp | name | value
--------------|----------|-------- 
2020-10-29 16:28:00 | foo | 18
2020-10-29 16:25:00 | foo | 1

This could be solved with a new transformation, or a simpler approach would be to add time functions to the existing 'add field from calculations' transformation.  You could then add a new field that quantizes each timestamp by the desired interval, then use the existing 'group by' transformation to aggregate data by each quantized timestamp.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by pablorf-dev** [comment]
I'm also interested on this feature.

I tried to do what documentation says [here](https://grafana.com/docs/grafana/latest/panels/transform-data/transformation-functions/#convert-field-type), but I've been unable to achieve it.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82995. Feel free to continue the discussion around this there. Thank you!
